core/types/transaction_test.go

```diff
@@ -217,5 +220,19 @@ func TestTransactionJSON(t *testing.T) {
 		if tx.ChainId().Cmp(parsedTx.ChainId()) != 0 {
 			t.Errorf("invalid chain id, want %d, got %d", tx.ChainId(), parsedTx.ChainId())
 		}
+		if tx.L1MessageSender() != nil && parsedTx.L1MessageSender() != nil && *tx.L1MessageSender() != *parsedTx.L1MessageSender() {
+			t.Errorf("invalid L1MessageSender, want %x, got %x", tx.L1MessageSender(), parsedTx.L1MessageSender())
+		}
+	}
+}
+
+// Tests that L1MessageSender has no impact on hash
+func TestL1MessageSenderHash(t *testing.T) {
```

Comment:
I see the logic above where you are enforcing this, but don't fully understand why--is it related to not breaking tests?

I feel like it shouldn't matter either way since these transactions will already be hashed on chain, and probably not with an RLP encoding.  But still confused why this is done.

Reply:
Yeah, this would break all tests with pre-computed hashes and interaction with any 3rd party tools (for instance, ethers throws saying that hash != expected hash). I'd be concerned if it's possible to have the same exact message but with a different L1 sender or an empty one, but the L1 sender will only be populated and submitted by our L1 subscriber, so the nonce, at a minimum, should be different, right? If so, then we can maintain compatibility without worrying about uniqueness.

Reply:
Ahhh okay yeah I see your point WRT compatibility.

Hadn't thought of nonces being different too, so good point there, I think you're right.  It would require that we always pass in the correct nonce in these transactions, which shouldn't be a problem.  Although it's. an interesting point that we don't currently expose that nonce to the execution context unless I'm mistaken.  Might want to add that 🤔